### PR TITLE
ddl: Make the test assert if it encounters an error

### PR DESCRIPTION
**Pull request by zimulala**
Make sure the `c` comes from `TestXXX`.
Make sure `TestCancelAddIndex` uses the corrected job's ID.

### CODE REVIEW

File: ddl/ddl_db_test.go [hunk 1]
```diff
@@ -1376,7 +1380,7 @@ func (s *testDBSuite) testRenameTable(c *C, storeStr, sql string) {
 	failSQL = fmt.Sprintf(sql, "test1.t2", "test1.t2")
 	s.testErrorCode(c, failSQL, tmysql.ErrTableExists)
 
-	s.tk.MustExec("drop table test1.t2")
+	s.tk.MustExec("drop database test1")
```
Comment (shenli) [inline ddl/ddl_db_test.go]: Why change this case?

File: ddl/ddl_db_test.go [hunk 2]
```diff
@@ -334,6 +332,7 @@ func (s *testDBSuite) TestCancelAddIndex(c *C) {
 			checkErr = errors.Trace(err)
 			return
 		}
+		jobIDs := []int64{job.ID}
```
Comment (shenli) [inline ddl/ddl_db_test.go]: Why this is better?

Comment (zimulala) [reply ddl/ddl_db_test.go]: It's accurate job‘s ID. 
Original test is failed but doesn't assert. So we can't find the job's ID is wrong.

File: ddl/ddl_db_test.go (hunk 1 above)
Comment (zimulala) [reply ddl/ddl_db_test.go]: Because Other tests will create this database. So we need clean it.

**Comment by coocood** [comment]
LGTM

**Pull request merged**